### PR TITLE
Add documentation for test timeout utils

### PR DESCRIPTION
**Pull request by agarwal-navin**
Added documentation to the `timeoutAwait` and `timeoutPromise` APIs that explain what they do and what the default bevaior is - they will wait until just before the test times out as per the mocha settings.

[AB#4644](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4644)

### CODE REVIEW

File: packages/test/test-utils/src/timeoutUtils.ts [hunk 1]
```diff
@@ -8,12 +8,13 @@ import { assert, Deferred } from "@fluidframework/common-utils";
...
 export const defaultTimeoutDurationMs = 250;
 
-// TestTimeout class manage tracking of test timeout. It create a timer when timeout is in effect,
-// and provide a promise that will be reject before the test timeout happen with a `timeBuffer` of 15 ms.
-// Once rejected, a new TestTimeout object will be create for the timeout.
-
 const timeBuffer = 15; // leave 15 ms leeway for finish processing
 
+// TestTimeout class that manages tracking of test timeout. It creates a timer when timeout is in effect,
+// and provides a promise that will be rejected before the test timeout happens with a `timeBuffer` of 15 ms.
```
Comment (alexvy86) [inline packages/test/test-utils/src/timeoutUtils.ts:14]:
```suggestion
// and provides a promise that will be rejected some time (as defined by `timeBuffer`) before the test timeout happens.
```

File: packages/test/test-utils/src/timeoutUtils.ts [hunk 2]
```diff
@@ -125,13 +126,72 @@ export type PromiseExecutor<T = void> = (
 	reject: (reason?: any) => void,
 ) => void;
 
+/**
+ * Wraps the given promise around another promise that will complete after a specific timeout if the original
```
Comment (alexvy86) [inline packages/test/test-utils/src/timeoutUtils.ts:130]:
```suggestion
 * Wraps the given promise with another promise that will complete after a specific timeout if the original
```

File: packages/test/test-utils/src/timeoutUtils.ts [hunk 3]
```diff
@@ -125,13 +126,72 @@ export type PromiseExecutor<T = void> = (
...
+): Promise<T> {
 	return Promise.race([promise, timeoutPromise<T>(() => {}, timeoutOptions)]);
 }
 
+/**
+ * Creates a promise from the given executor that will complete after a specific timeout if the original
+ * promise does not resolve by then. By default, it will use the mocha test timeout and complete the promise
+ * just before that so that tests don't time out because of unpredictable awaits.
+ * The timeout can be overridden via timeoutOptions but it's recommended to use the default value.
+ * @param promise - The promise to be awaited.
```
Comment (alexvy86) [inline packages/test/test-utils/src/timeoutUtils.ts:152]: Should be `executor` in this one.

**Review by alexvy86 — APPROVED** [review]
A few nits, but overall lgtm.

**Review by curtisman — APPROVED** [review]

**Pull request merged**